src/picomsg/src/oper_db_writer.cpp

```diff
@@ -114,3 +190,282 @@ bool oper_db_writer_set_terminator_vrf_tunnel_counter(const uint32_t terminator_
+		return false;
+	}
+
+	if (! (*oper_db_client)->set((*oper_db_client).get(), db_obj.release(), 0)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to write vrf[%s] device-id[%s] tunnel oper data to oper DB", oper_data->vrf_name, oper_data->device_id);
+		return false;
+	}
+
+	db_obj = oper_db_client_create_db_object(*oper_db_client, LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA, keys);
+	if (! db_obj) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to create oper DB object for class ID LIBOPER_DB_CLIENT_CLASS_ID_TUNNEL_OPER_DATA");
+		return false;
+	}
+
+	if (! db_obj->enum_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_STATUS, oper_db_status_value)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_STATUS to %d",
+					oper_data->status);
+		return false;
+	}
+
+	if (! db_obj->u8_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TERMINATOR_ID, oper_data->term_id)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_TERMINATOR_ID to %hu",
+					oper_data->term_id);
+		return false;
+	}
+
+	if (! db_obj->enum_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACTIVENESS_STATUS,	oper_data->activeness_status ? LIBOPER_DB_CLIENT_ACTIVENESS_STATUS_ACTIVE :
+																																		LIBOPER_DB_CLIENT_ACTIVENESS_STATUS_STANDBY)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object enum LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACTIVENESS_STATUS to %d",
+					oper_data->activeness_status);
+		return false;
+	}
+
+	if (! db_obj->string_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACCOUNT_ID, oper_data->account_id)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_ACCOUNT_ID to %s",
+					oper_data->account_id);
+		return false;
+	}
+
+	if (! db_obj->string_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_NEIGHBOR_SUB_INTERFACE, oper_data->neigh_sub_interface)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_NEIGHBOR_SUB_INTERFACE to %s",
+					oper_data->neigh_sub_interface);
+		return false;
+	}
+
+	if(! version_to_oper_db(oper_data->version, oper_db_ike_version)) {
+		return false;
+	}
+
+	if (! db_obj->enum_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VERSION, oper_db_ike_version)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_VERSION to %u",
+					oper_data->version);
+		return false;
+	}
+	if (! db_obj->u16_set(db_obj.get(), LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DH_GROUP, oper_data->dh_group)) {
+		PICOMSG_LOG(PICOMSG_LOG_ERR, "failed to set oper DB object LIBOPER_DB_CLIENT_FIELD_ID_TUNNEL_OPER_DATA_DH_GROUP to %hu",
+					oper_data->dh_group);
+		return false;
+	}
+
+	enum LIBOPER_DB_CLIENT_FIELD_ENUM_VALUES oper_db_enc_algo, oper_db_int_algo;
```

Comment:
Define `oper_db_int_algo` only when needed - Right before the call to `int_algo_to_oper_db()`